src/filesystem/interface.js

```diff
@@ -206,6 +206,50 @@ function FileSystem(options, callback) {
     return watcher;
   };
 
+  //Object that uses filenames as keys
+  const statWatchers = new Map(); 
+
+  this.watchFile = function(filename, options, listener) {
+    let prevStat, currStat; 
+
+    if (Path.isNull(filename)) {
+      throw new Error('Path must be a string without null bytes.');
+    }
+    // Checks to see if there were options passed in and if not, the callback function will be set here
+    if (typeof options === 'function') {
+      listener = options; 
+      options = {};
+    }
+    // default 5007ms interval, persistent is not used this project
+    const interval = options.interval || 5007;
+    listener = listener || nop;
+    
+    // Stores initial prev value to compare 
+    fs.stat(filename, function(err, stats) { 
+      prevStat = stats;
+      
+      // Stores interval return values
+      statWatchers.set(filename, value);
+      
+      var value = setInterval(function() {
```

Comment:
We need to cache this `value` so we can get at it later in `unwatchFile` to cancel the interval.

Reply:
I thought setting the `value` as a value to the key `filename` serves this purpose. If not, can caching be done in the local storage? I haven't cache anything before, but I see that there are multiple ways to cache things.

Reply:
@humphd I thought setting the value as a value to the key filename serves this purpose. If not, can caching be done in the local storage? I haven't cache anything before, but I see that there are multiple ways to cache things.

Reply:
But look at your code above.  You are setting `value` on 232 as the value in your map, but it doesn't actually have it's real value yet until 234.  Maybe just move line 232 down below the `setInterval`